templates/flavors/nvidia-gpu/patches/kubeadm-controlplane.yaml

```diff
@@ -0,0 +1,10 @@
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+kind: KubeadmControlPlane
+metadata:
+  name: "${CLUSTER_NAME}-control-plane"
+spec:
+  kubeadmConfigSpec:
+    useExperimentalRetryJoin: true
+    postKubeadmCommands:
+      # Install the NVIDIA device plugin for Kubernetes
+      - KUBECONFIG=/etc/kubernetes/admin.conf kubectl apply -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/master/nvidia-device-plugin.yml
```

Comment:
why run the nvidia install script on the control plane nodes if only the worker nodes are GPU enabled?

Reply:
Because the manifest is just a daemonset that will schedule itself on GPU agent nodes, so it could be installed from anywhere (the original version of this PR had the user do it), and because I know I have `kubectl` and a kubeconfig on the master, but I don't think that's true on the agent nodes.